### DISCUSSION TITLE
Allow modification of certain user fields only by the administrator and set the default value

### DISCUSSION THREAD

**Discussion by Scylin232** [pocketbase/pocketbase#1622] [Q&A] [answered]
Hello, thank you very much for such an amazing project! 

I'm trying to figure out how I can allow only the admin to modify a certain field and give it a default value. 

Example: https://imgur.com/eZgvozu

The Coins value is important for security and I don't want anyone on my frontend to set it to an arbitrary value, or modify it via the API, and it would be ideal to set it to a default value from the Pocketbase admin panel. Is that possible?

**Comment by ganigeorgiev** [comment, marked as answer]
> I'm trying to figure out how I can allow only the admin to modify a certain field 

In general there are 3 options:

1. Move your private/protected fields in a separate collection with its own API rules (for example you can allow only list and view access to the coins) and link the new collection with a `relation` field to your main collection.

2. Add an API rule to disallow submitting the `coins` field. With [v0.11+](https://github.com/pocketbase/pocketbase/releases/tag/v0.11.0) there is now support for the `:isset` field modifier:
    ```js
    // prevent users to submit any "coins" value (including null or zero-value)
    @request.data.coins:isset = false
    ```
    Alternatively, you can also request a "fixed" default value to be always send with the create request by using a filter rule like:
    ```js
    @request.data.coins = 123
    ```

3. Manually exclude/validate/overwrite/etc. the submitted value in a before event hook(s) - https://pocketbase.io/docs/event-hooks/.


>  it would be ideal to set it to a default value from the Pocketbase admin panel

PocketBase doesn't have default values because we don't store `null` (_with exception of the `json` type field_) and uses zero-defaults and it is ambiguous when it will be applied and for now it is left to the developers to handle the defaults on their own by sending the default client-side or handling it in a server-side [event hook](https://pocketbase.io/docs/event-hooks/). For example:

```go
app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
	if e.Record.Collection().Name != "users" {
		return nil // not a user
	}

	// set default coints value
	e.Record.Set("coins", 123)

	return nil
})
```

**Reply by Scylin232** [reply]
Your love and attention to this project is beyond praise, thank you very much! Especially for the help in such specific questions.
